src/main/java/ch/wisv/areafiftylan/products/service/OrderServiceImpl.java

```diff
@@ -116,25 +111,41 @@ public Order addTicketToOrder(Long orderId, TicketDTO ticketDTO) {
             throw new IllegalStateException("Order limit reached");
         }
 
-        User user = order.getUser();
-
         // Request a ticket to see if one is available. If a ticket is sold out, the method ends here due to the
         // exception thrown. Else, we'll get a new ticket to add to the order.
-        Ticket ticket = ticketService
-                .requestTicketOfType(ticketDTO.getType(), user, ticketDTO.hasPickupService(), ticketDTO.isCHMember());
-
+        Ticket ticket;
+        if (order.getUser() != null) {
+            ticket = ticketService.requestTicketOfType(order.getUser(), type, pickupService, chMember);
+        } else {
+            ticket = ticketService.requestTicketOfType(type, pickupService, chMember);
+        }
         order.addTicket(ticket);
         return orderRepository.save(order);
     }
 
     @Override
-    public Order removeTicketFromOrder(Long orderId, TicketDTO ticketDTO) {
-        Order order = orderRepository.getOne(orderId);
-        if (order.getStatus().equals(OrderStatus.CREATING)) {
+    public Order assignOrderToUser(Long orderId, String username) {
+        Order order = getOrderById(orderId);
+        User user = userService.getUserByUsername(username);
+
+        if (order.getStatus() != OrderStatus.ANONYMOUS) {
+            throw new ImmutableOrderException("Order already assigned!");
+        }
+
+        order.setUser(user);
+        order.setStatus(OrderStatus.ASSIGNED);
+        return orderRepository.save(order);
+    }
+
+    @Override
+    public Order removeTicketFromOrder(Long orderId, TicketType type, boolean pickupService, boolean chMember) {
+        Order order = getOrderById(orderId);
+        if (order.getStatus().equals(OrderStatus.ANONYMOUS) || order.getStatus().equals(OrderStatus.ASSIGNED)) {
 
             // Find a Ticket in the order, equal to the given DTO. Throw an exception when the ticket doesn't exist
-            Ticket ticket = order.getTickets().stream().filter(isEqualToDTO(ticketDTO)).findFirst()
-                    .orElseThrow(TicketNotFoundException::new);
+            Ticket ticket =
+                    order.getTickets().stream().filter(isEqualToInput(type, pickupService, chMember)).findFirst()
+                            .orElseThrow(TicketNotFoundException::new);
 
             order.getTickets().remove(ticket);
             ticketService.removeTicket(ticket.getId());
```

Comment:
This has to do with the one-way relation between Orders and Tickets. This can't be done automatically unless a Ticket gets a referenced order. At least that was my conclusion last time I looked at it.